cpu/native/include/periph_cpu.h

```diff
@@ -96,6 +96,52 @@ typedef enum {
 } gpio_flank_t;
 
 /** @} */
+#else
+
+/**
+ * @brief   Mocked GPIO
+ *
+ * Mocked GPIO representation for simulation.
+ * @{
+ */
+typedef struct {
+    int value;              /**< current value */
+    int mode;               /**< current mode */
+    int flank;              /**< flank to trigger interrupts */
+    void (*cb)(void *arg);  /**< ISR */
+    void *arg;              /**< ISR arg */
+} gpio_mock_t;
+/** @} */
+
+#define GPIO_UNDEF          0
+
+#ifndef GPIO_PORT_MAX
+#define GPIO_PORT_MAX       (16)
+#endif
+
+#ifndef GPIO_PIN_MAX
+#define GPIO_PIN_MAX        (32)
+#endif
+
+/**
+ * @brief Mocked GPIO array
+ */
+static gpio_mock_t __attribute__((unused)) gpio_mock[GPIO_PORT_MAX][GPIO_PIN_MAX];
```

Comment:
this array needs to go into the c file and this needs to be
```suggestion
extern gpio_mock_t gpio_mock[GPIO_PORT_MAX][GPIO_PIN_MAX];
```

else we have an array per compilation unit

which is probably not intended

Reply:
Done